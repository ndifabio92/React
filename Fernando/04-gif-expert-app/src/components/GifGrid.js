// import React, { useState, useEffect } from 'react';
import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGitfs';
import { GifGridItem } from './GifGridItem';
import PropType from 'prop-types';
// import { getGifs } from '../helpers/getGifs';

export const GifGrid = ({ category }) => {

    // const [count, setCount] = useState(0);
    // const [ images, setImages ] = useState([]);
    const { data:images, loading } = useFetchGifs( category );
    // useEffect( () => {
    //     getGifs( category )
    //         .then( setImages );
    // }, [ category ] );

    return (
        <>
            <h3 className="animate__animated animate__bounce animate__delay-2s">{ category }</h3>
            { loading && <p className="animate__animated animate__bounce animate__flash">Loading</p> }
            <div className="card-grid">
                {
                    images.map( ( img ) => (
                        <GifGridItem 
                            key = { img.id } 
                            { ...img } // img = { img }
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropType.string.isRequired
}