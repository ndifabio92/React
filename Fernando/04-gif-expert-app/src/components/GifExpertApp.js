import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {
    // const categories = ['one punch','samurai x','dragon ball'];

    const [ categories, setCategories ] = useState(['']);

    // const handleAdd = () => {
    //     setCategories( cats => [ ...cats, 'HunterXHunter']);
    // }

    return (
        <>
            <h2>GrifExpertApp</h2>
            <AddCategory setCategories = { setCategories }/>
            <hr/>
            {/* <button onClick={ () => handleAdd() }>Agregar</button> */}
            <ol>
                {/* { 
                    categories.map( category => {
                        return <li key={ category }> { category} </li>
                    })
                } */}
                {
                    categories.map( category => (
                        <GifGrid key = { category } category = { category } />
                    ))
                }
            </ol>
        </>
    )
}
