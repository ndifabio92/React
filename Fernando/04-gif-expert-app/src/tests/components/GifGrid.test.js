import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGitfs';
jest.mock('../../hooks/useFetchGitfs');

describe('Pruebas en el <GrifGrid/>', () => {

    const category = 'One Punch';

    test('debe de mostrarse correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category = { category }/> );
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category = { category }/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
})