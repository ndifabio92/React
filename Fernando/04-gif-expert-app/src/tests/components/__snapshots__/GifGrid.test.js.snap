// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GrifGrid/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__delay-2s"
  >
    One Punch
  </h3>
  <p
    className="animate__animated animate__bounce animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el <GrifGrid/> debe mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__delay-2s"
  >
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;
